src/api/app/controllers/webui/requests/role_additions_controller.rb

```diff
@@ -0,0 +1,43 @@
+module Webui
+  module Requests
+    class RoleAdditionsController < Requests::MainController
+      before_action :require_login
+      before_action :lockout_spiders
+      before_action :set_package, only: [:new]
+      before_action :set_project, only: [:new]
+
+      after_action :verify_authorized
+
+      def new
+        # TODO: Don't use request_action_attributes here since it relies on params
+        authorize BsRequest.new(bs_request_actions: [BsRequestAction.new(request_action_attributes(:add_role))]), :create?
+      end
+
+      def create
+        request = BsRequest.new(description: params[:add_role_description],
+                                bs_request_actions: [BsRequestAction.new(request_action_attributes(:add_role))])
+        authorize request, :create?
+        begin
+          request.save!
+        rescue APIError => e
+          flash[:error] = e.message
+          redirect_to(controller: :package, action: :show, package: params[:package_name], project: params[:project_name]) && return if @package
+          redirect_to(controller: :project, action: :show, project: params[:project_name]) && return
+        end
+        redirect_to request_show_path(request.number)
+      end
+
+      private
+
+      def set_package
+        return unless params.key?(:package_name)
+
+        @package = Package.find_by(name: params[:package_name])
+      end
+
+      def set_project
+        @project = @package.try(:project) || Project.find_by(name: params[:project_name])
+      end
```

Comment:
We're doing this since the controller is a resource shared between projects and packages.